### ISSUE TITLE
Need guidance on prerequisites

### ISSUE THREAD

**Issue by stoobie** [redhat-documentation/supplementary-style-guide#67]
The IBM Style Guide does not include guidelines for prerequisites.

The style council discussed guidance for writing prerequisites in documentation, but no PR was created yet. This issue is just for tracking.

This issue recently popped up as a discussion on the modular documentation gChat, and I realized that we need to create a PR.

The style council, after much discussion and dissension, decided to write prerequisites in the passive voice. The reasoning: Active voice turns prerequisites into procedural steps. E.g. if you want the use to be logged in with Admin permissions when the procedure begins:

**Passive:** You are logged in with Administrator privileges
vs
**Active:** Log in with Administrator privileges.

**Comment by oraNod** [comment]
My 2c on all of this is that using the imperative voice doesn't make something a procedure. A procedure is a set of steps. 

The imperative structure keeps an authoritative style and is clear. It also addresses the user directly, even though the "you" is implied.

I get the idea that pre-reqs are a sort of "checklist" for things that need to be complete before diving into a procedure. However, I think using passive voice and repetitive sentences "You have ..." makes our docs harder to read. If pre-reqs are not concise, users are likely to skim over them to get to the procedure.

Part of the issue with the imperative structure might be the question as to whether the pre-req should be part of the actual procedure. In the case of the example in the description for this PR, "Log in with Administrator privileges.", that isn't something that needs to be done upfront. Logging in is a trivial step and would not interrupt the flow of the procedure.

Pre-reqs should be significant steps or required artifacts or pieces of information that the user needs to gather before starting the procedure. You want users to be able to complete the steps in a procedure without getting distracted or going off task.

Continuing with the example of having admin permissions, that is a requirement. Users might also need to request admin permissions from a system, which we all know can take a long time. And users cannot log in with admin priviledges if they don't have them. So I would argue that the pre-req is something like "Ensure you have admin privileges." then the first step of the procedure can be "Log in ..."

**Comment by ndeevy** [comment]
I like the idea of helping writers write good prereqs but I'm a bit unsure about including the passive at all and the wording of this doesn't take everything into account.

Passive voice removes or hides the agent of the action but the agent is clear in this sentence (`You are logged in with root permissions.`). This sentence just describes a state in the present tense so I think it might not be helpful confuse this with passive voice as we usually talk about it. And, I don't think it actually _is_ the passive voice, despite the verb form 'are logged' being part of it.

Another way to write the same prereq without talking about passives would be to recommend using the present perfect tense:
`*You have logged in with root permissions.`
This is also just as wordy though. 

Also, some prereqs are not sentences and I think that should be fine. For example,
`.Prerequisites`
`* A functioning X deployment.`
`* Log in credentials.`
But this guidance suggests that all prereqs should be sentences in a passive construction.

Using the already-covered-in-IBM rules about keeping lists parallel whenever you can, would seem to fix most problems I've ever seen with prereqs... <--- possibly an unpopular opinion though! :-P

**Comment by stoobie** [comment]
@ndeevy @oraNod Good comments.
@bergerhoffer  @cbudz @emmurphy1 Do you have input?

**Comment by ncbaratta** [comment]
I believe we should avoid passive voice wherever possible and vote for active voice in prerequisites.

**Comment by bergerhoffer** [comment]
I honestly don't have a strong opinion on this, and I can't actually remember the prior discussion. I see pros and cons to both ways. I personally tend to write prereqs in the passive voice, but I definitely get the desire for active voice when possible. My question here would be how to word in active voice when the prerequisite isn't done by the same person as the person performing the procedure. Or maybe has been done in the past awhile ago (like taking a backup), and not right before the procedure is done.

I guess my point is that maybe we need to brainstorm the different types/examples of prereqs you can have, and make sure that they work for whatever style is chosen, if we need to add guidance.

**Comment by IngridT1** [comment]
Sometimes, you have to use passive voice in a prerequisite or in technical writing in other areas. But, in general, I would avoid it, if at all possible. 

If we put this out as a guideline to the broader writing community, I think that you would see a lot of people who strongly object to encouraging passive voice in any way.

**Comment by ndeevy** [comment]
We have this general rule (ISG) to use active voice as much as possible. Most people seem to follow that when writing so I think _maybe_ we don't need to mention voice here at all.

Making this into guidance more like:
  'your prereqs don't necessarily have to be parallel.
   <examples>'  
seems more helpful.

**Comment by sterobin** [comment]
So to summarize some thoughts on our CCS style council today in favor of passive voice ...

The purpose of prerequisites is by nature to list things that should already be present or completed before you begin the procedure that follows, i.e., they're checks not tasks. If they're not complete, go and do them, if they are, then press on. If we put a prereq in active voice, then it's no longer a check but a task, in which case it's no longer a prerequisite but a step or maybe even it's own procedure.

Also, some or even many prerequisites aren't necessarily completed by the current user. For example, having a instance running or having something installed on a system or having a configuration enabled are key requirements for X procedure that should have been done previously and likely by a system admin and perhaps not by the user at hand. The current user would see that prereq, and if not sure and either check or confirm with the relevant admin user.

So imo, the vast majority of prereqs are naturally non-active statements of current states of previous actions.

I know there's debate about this, so consider the following prerequisites, which are in **non-active statements**.
 
```
.Prerequisites
* https://adoptopenjdk.net/[JDK^] 11 or later is installed.
* You have a running Kafka instance in {product}.
* You are logged in to the X Portal.
```

First prereq: This is something that should already be installed, whether done by the user or the person who set up the user's system is irrelevant. But it needs to be there.

Second prereq: This user her/himself needs to have a running Kafka instance. It can't be any old instance by another user, but specifically one they have created and run, which was described in the previous procedure as well. So must be the current user and must already have been done, which in the context of the doc is quite logical.

Third prereq: Also must be done already and by the current user (obviously), and is necessary to mention because you can achieve similar tasks in this procedure using the CLI instead of the Portal, so we're clarifying that the user must be logged in to navigate through where they need to be. And again it's logical given the context so is a sign-posting check point.

Now compare the same prerequisites in active voice, which I still don't really understand:

```
.Prerequisites
* Install https://adoptopenjdk.net/[JDK^] 11 or later.
* Create a Kafka instance in {product}.
* Log in to the X Portal.
```

Note how the meaning and intent is now changed considerably. They or another user would've already done all of these, so we've now made them second guess themselves and even confused them. Also note that when we repeat prereqs like these, it's strange to tell them to do the same thing multiple times, wherase with non-active statements, it's fine to repeat the same checks because we're just saying the same states apply multiple places.

All of this in mind, I personally don't following the argument for using active/imperative voice in prerqs, since it contradicts the very nature of them. Therefore, my recommendation would be that prereqs should be non-active statements that should be true in order to proceed with a procedure, whether that means using passive voice or simply non-active formations of other kinds.

**Comment by stoobie** [comment]
Thank you @sterobin for that very clear explanation. 

The bottom line that I took away from the meeting is: _Be as clear and concise as possible. You can use the passive voice if necessary to achieve that end._

Based on all the opinions expressed, I suggest the following guidelines:

Prereqs should be non-active statements that should be true in order to proceed with a procedure. They are checks not tasks. Describe them clearly as something that is already completed. 
- As always, be as clear and concise as possible.
- Use non-imperative formations. You can use the passive voice if necessary to be more clear and concise.
- Keep lists parallel whenever you can, but if necessary, you can use different tenses in different prerequisites

**Examples**

DO:

* A functioning X deployment.
* Log-in credentials.
* admin privileges.
* https://adoptopenjdk.net/[JDK^] 11 or later is installed.
* You have a running Kafka instance in {product}.
* You are logged in to the X Portal.

DO NOT
* Deploy X
* Log in to the X Portal.
* Request admin privileges
* Install JDK 11 or later

@oraNod @ncbaratta @ndeevy Can any of you provide other examples? Or examples that contradict this?

It seems that there's going to be a need for some to "Disagree and commit".

**Comment by ndeevy** [comment]
Clearly explained! Thanks @sterobin, a lot of that makes much more sense to me now.
@stoobie your summary makes sense: _Be as clear and concise as possible. You can use the passive voice if necessary to achieve that end._

One thing - I feel like the term "non-active" might introduce some confusion when we talk about active and passive so maybe we could tweak that a bit. From the above examples: 

- `You have logged in to the portal.`  (S-V-O) is present perfect tense (past) in active voice because the agent is not hidden.
- `You are logged in to the portal.` (S-V-O) is present tense and also active not passive because the agent is clear.
- True passive would be something weird like 'The portal is logged in to <by you>' (O-V-< S >) with the agent optionally hidden.
- `https://adoptopenjdk.net/[JDK^] 11 or later is installed.` is the only real passive and it's a bonafide use of passive (the agent is unknown or unimportant). 

So because of the pre-existing rule to use active voice whenever possible (that we don't want to discourage I _think_ judging by the previous conversations), I think we could avoid the term "non-active" because it might be misinterpreted and active/passive is already tricky. Like Steve suggested, saying 'you can use passive if necessary' covers when it's legit needed, like that one:  `..([JDK^] 11 or later is installed.` 

Would an edit along these lines include everything?: 

Write prerequisites as checks that the reader must have completed or that are true before they begin a procedure. They can be actions that the reader, another person, or piece of technology has completed. Prerequisites can also include items that the reader must have ready before they begin the procedure steps.`
-     Be as clear and concise as possible.
-     Do not use imperative formations.
-     Use the passive voice if necessary to be more clear and concise.
-     You can write unparallel prerequisite lists if necessary for clarity.


One other question: should we include in this guidance that we must use the **.Prerequisites** heading even if there's only one prereq or is that something that's really covered as part of moddocs guidance?

**Comment by sterobin** [comment]
@stoobie @ndeevy , all makes sense to me. I'm not sure I'm totally convinced on the unparallel aspect. In @stoobie 's examples, what I do in reality is "You have a functioning ...", "You have login credentials ...", etc. I know Steve is specifically saying that he's not in favor of that due to the repetition, but to me it's preferable and less awkward than the repetition. Plus it gets confusing then why the prereq like "You have a running instance ..." wouldn't just be "A running instance ..." I as the writer know that it's important that the running instance be specifically yours as the user and no one else's, but the unparallel structure just looks more like a mistake than an intentional emphasis.

I'll leave the parallelism to whatever others thing, but imo I prefer parallel repetition.

And @ndeevy , the `.Prerequisites` for one prereq is already covered by the mod docs resources, so fine if we do or don't want to mention it I think.

**Comment by IngridT1** [comment]
@ndeevy @sterobin When I said that your lists didn't have to be parallel, I meant that you didn't have to worry if some of the complete sentences were in passive voice and others were in active voice. Heck, if you wanted to have an imperative sentence in the list, I think that would still be parallel, as long as they were all complete sentences. Parallelism, in general, is a good idea, but it doesn't have to be precise. 

As far as the guideline "Do not use imperative formations." goes, do we have buy-in across CCS? I certainly have seen a list of imperatives for Prerequisites. They are generally more along of the lines of:

- Ensure that you have a Kafka instance. 

I know that Laura Bailey has an opinion about using imperatives.

**Comment by jlprevatt** [comment]
I stick by my thoughts in #68 that something like the following construction (lead-in sentence, grocery list, extra info as needed) eliminates a lot of the problems with active/passive, parallel construction, and who does the action:

Prerequisites

Before you begin, you must meet the following prerequisites:

- This thing.
- That thing.
- Another thing. You need this thing because why. This takes as long as it takes to explain.
- One more thing. This thing was probably installed and configured by your database/system administrator; confirm with that person.
